app/controllers/spaces_controller.rb

```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
 
   #GET /spaces
   #GET /spaces.json
+  #TODO(matthew): params['strings'] or params[:symbols]?
   def index
     if params['user_id']
       @user = User.find(params['user_id'])
-      @spaces = @user.spaces.visible_by(current_user)
+      if current_user && @user.id == current_user.id
+        @spaces = @user.spaces
+      else
+        @spaces = @user.spaces.is_public
+      end
```

Comment:
It's not a scope anymore, its a class method to account for organizations. We could use a renamed scope here, but this way seemed cleaner to me anyways, less double scoping. Though maybe rails makes the sql the same anyhow.